student.c

```diff
@@ -1,20 +1,42 @@
-#include <assert.h>
 #include "student.h"
 
+#include <assert.h>
+
+static ll_comparator_t compare_gpa(struct ll_node *first,
```

Comment:
This declaration works because C pointers are mostly implicitly convertable, but I'll note that you're actually stating that this function returns an `ll_comparator_t`, which is really just a function pointer under the hood.  To accurately reflect the definition of `ll_comparator_t`, you should declare this function to return `int` instead.